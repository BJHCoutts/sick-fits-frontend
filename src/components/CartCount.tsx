import styled from 'styled-components'
import { TransitionGroup, CSSTransition } from 'react-transition-group'

export default function CartCount({ count }: { count: number }) {
	return (
		<SAnimation>
			<TransitionGroup>
				<CSSTransition
					unmountOnExit
					className="count"
					classNames="count"
					key={count}
					timeout={{ enter: 400, exit: 400 }}
				>
					<SDot>{count}</SDot>
				</CSSTransition>
			</TransitionGroup>
		</SAnimation>
	)
}

const SDot = styled.div`
	background-color: var(--red);
	color: white;
	border-radius: 50%;
	padding: 0.5rem;
	line-height: 2rem;
	min-width: 3rem;
	margin-left: 1rem;
	font-variant-numeric: tabular-nums;
`
const SAnimation = styled.span`
	position: relative;
	.count {
		display: block;
		position: relative;
		transition: transform 0.4s;
		backface-visibility: hidden;
	}
	.count-enter {
		transform: scale(4) rotateX(.5turn)
	}
	.count-enter-active {
		transform: rotateX(0);
	}
	.count-exit {
		top: 0;
		position: absolute;
		transform: rotateX(0);
	}
	.count-exit-active {
		transform: scale(4) rotateX(.5turn):
	}
`
