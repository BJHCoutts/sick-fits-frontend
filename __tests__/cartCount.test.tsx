import { render } from '@testing-library/react'
import CartCount from '../src/components/CartCount'

describe('<CartCount />', () => {
	it('renders', () => {
		render(<CartCount count={10} />)
	})
	it('matches snapshot', () => {
		const { container } = render(<CartCount count={12} />)
		expect(container).toMatchSnapshot()
	})
	it('updates via props', () => {
		const { container } = render(<CartCount count={12} />)
		expect(container.textContent).toBe('12')
		expect(container).toHaveTextContent('12')
	})
})
