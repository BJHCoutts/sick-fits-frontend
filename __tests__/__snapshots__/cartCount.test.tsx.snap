// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> matches snapshot 1`] = `
<div>
  <span
    class="sc-gsTDqH WQfJr"
  >
    <div>
      <div
        class="sc-bdfCDU iBUbSz count"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
